### PR TITLE
Rearrange components and carriers

### PR DESCRIPTION
**Pull request by jnnr**
This PR:

* Replaces the carrier 'heat' and all connected components with 'heat_central' and 'heat_decentral'. 
* Adds a component h2-bpchp.
* Adds a script that helps to create templates for input data from a scenario definition

### CODE REVIEW

File: scenarios/base-scenario.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+links:
+  - BB-B
+
+busses:
+  - ch4
+  - electricity
+  - h2
+  - heat_central
+  - heat_decentral
+  - biomass
```
[the remark this answers is not in the document]
Comment (jnnr) [reply scenarios/base-scenario.yml:21]: Ok. Biomass is an currently implemented as an unbalanced bus. We could drop it and implement biomass-gt as a source. Or leave it as it is.

File: scenarios/base-scenario.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+name: base-scenario
+
+regions:
+  - BB
+  - B
+
+links:
+  - BB-B
+
+busses:
```
[the remark this answers is not in the document]
Comment (jnnr) [reply scenarios/base-scenario.yml:15]: We will cover the methanation soon.

File: oemof_b3/model/component_attrs_update.yml [hunk 3]
```diff
@@ -14,4 +14,115 @@ electricity-electrolyzer:
...
+heat_central-shortage:
+  carrier: heat_central
+  tech: shortage
+  type: shortage
+  foreign_keys:
+    bus: heat_central
+  defaults:
+    output_parameters: "{}"
+
+heat_central-storage:
```
[the remark this answers is not in the document]
Comment (jnnr) [reply oemof_b3/model/component_attrs_update.yml:45]: Actually, it's not - storage vs. shortage?

File: scenarios/base-scenario.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  - heat_central
+  - heat_decentral
+  - biomass
+
+components:
+  - biomass-gt
+  - ch4-boiler
+  - ch4-bpchp
+  - ch4-demand
+  - ch4-excess
```
[the remark this answers is not in the document]
Comment (jnnr) [reply scenarios/base-scenario.yml:28]: Excess and shortage translate to export and import, demand is something else. So shortage is missing. Resolved in ffbe395

File: scripts/build_datapackage.py [hunk 5]
```diff
@@ -170,6 +201,8 @@ def parametrize_sequences(edp, ts, filters):
 
     scalars = load_b3_scalars(path_scalars)
 
+    scalars = expand_regions(scalars, scenario_specs["regions"])
+
```
[the remark this answers is not in the document]
Comment (jnnr) [reply scripts/build_datapackage.py:223]: Done in 2ab077c

**Comment by jnnr** [comment]
> I understand the process to be the following (would be good to document in rtd)

That's great! Where would you include this in the docs? In the chapter "model pipeline", or in the header of `build_datapackage.py`?

**Comment by SabineHaas** [comment]
> > I understand the process to be the following (would be good to document in rtd)
> 
> That's great! Where would you include this in the docs? In the chapter "model pipeline", or in the header of `build_datapackage.py`?

I think there should be a general explanation on how oemof-B3 works - so I guess "model pipeline" or even something like "general structure"?

**Comment by jnnr** [comment]
The script "create_empty_scalars.py" is cleaned up a bit. It is now simpler to create empty oemof_b3-scalars either in raw or resources (where annuisation already combined the cost, lifetime and wacc). 

Using the script, I created empty raw scalars for the basic scenario, set all values to zero and added a value for the wacc (on the fileserver now), annuised it with prepare scalars and fed it into the base-scenario defined in #69. It works, which means that the optimization script succeeds and passes the problem to the solver. 

This is one step towards feeding real data to the model. Next step is to check which real assumptions are missing in the raw scalars and complete them.

**Pull request merged**